test/mitmproxy/addons/test_mapeditor.py

```diff
@@ -0,0 +1,193 @@
+import tempfile
+import os
+import pytest
+
+from mitmproxy.test import tflow
+from mitmproxy.test import taddons
+
+from mitmproxy.addons import mapeditor
+
+
+class TestMapEditor:
+    def test_parse_mapeditor(self):
+        x = mapeditor.parse_mapeditor("~u .*://example.com/script.js:MAP_TO:/etc/hostname")
+        assert x == ("~u .*://example.com/script.js", "/etc/hostname")
+        with pytest.raises(Exception, match="Invalid map editor specifier"):
+            mapeditor.parse_mapeditor("~u .*://example.com/script.js:FOO_TO:/etc/hostname")
+
+    def test_configure(self):
+        me = mapeditor.MapEditor()
+        with taddons.context(me) as tctx:
+            with pytest.raises(Exception, match="Invalid map editor filter pattern"):
+                tctx.configure(me, mapeditor = ["~b:MAP_TO:/etc/hostname"])
+            tctx.configure(me, mapeditor = ["foo:MAP_TO:/etc/hostname"])
+
+    def test_mapeditor_matching(self):
+        me = mapeditor.MapEditor()
+        with taddons.context(me) as tctx:
+            # test matching rules
+            text_test_fp = os.path.join(tempfile.gettempdir(), os.urandom(24).hex())
```

Comment:
pytest has a wonderful `tmp_path` fixture which also takes care of cleaning up: https://docs.pytest.org/en/latest/tmpdir.html

Reply:
On a second note: I don't think we need to create temp files at all, you can use have the resources in the test directory. See https://github.com/mitmproxy/mitmproxy/blob/master/test/mitmproxy/test_certs.py#L151 for example.